clients/pkg/promtail/targets/file/filetargetmanager.go

```diff
@@ -356,15 +355,38 @@ func (s *targetSyncer) sync(groups []*targetgroup.Group, targetEventHandler chan
 		}
 	}
 
+	s.droppedTargets = dropped
+
+	// keep track of how many targets are using a fileEventWatcher
+	watcherUseCount := make(map[string]int, len(s.fileEventWatchers))
+	for _, target := range s.targets {
```

Comment:
we could simplify/remove the need for this extra loop on every `sync` call by storing the `watcherUseCount` as a field in the `targetSyncer` struct